src/sparseml/pytorch/torchvision/train.py

```diff
@@ -323,9 +324,20 @@ def collate_fn(batch):
         steps_per_eval = min(steps_per_eval, args.max_eval_steps)
 
     print("Creating model")
-    model = torchvision.models.get_model(
-        args.model, weights=args.weights, num_classes=num_classes
-    )
+    if args.arch_key in ModelRegistry.available_keys():
+        with torch_distributed_zero_first(args.rank if args.distributed else None):
```

Comment:
it sounds to me that by passing rank = None you run in non-parallel (synchronous?) mode. it would be probably more readable to not use the context manager here if the synchronous mode is chosen. otherwise confusing.